src/schemathesis/runner.py

```diff
@@ -0,0 +1,21 @@
+from typing import Callable
+
+import requests
+
+from .models import Case
+from .schemas import BaseSchema
+
+
+def execute(url: str, schema: BaseSchema) -> None:
+    def single_test(case: Case, check: Callable) -> None:
+        response = requests.request(
+            case.method, f"{url}{case.formatted_path}", headers=case.headers, params=case.query, json=case.body
+        )
+        check(response)
+
+    for _, test in schema.get_all_tests(single_test):
```

Comment:
Maybe this part would be much faster if we used `async`. Instead of using `requests`, we could maybe use `aiohttp.ClientSession`?

Reply:
My initial try to implement it with aiohttp failed, unfortunately :( It would be nice, yep.

The problem was, that the test itself is sync (however there are some ways to fix it via hypothesis hooks - https://hypothesis.readthedocs.io/en/latest/details.html?highlight=executor#custom-function-execution)
which means that there are multiple async tasks that execute blocking code which doesn't bring benefit - thread/process pool can fix this, but probably it would be better to go with non-blocking tasks (that executor thing above).

I'd be very happy to go in this direction and appreciate any help :)

Reply:
I see. I will check it when I have time :-). I don't know async very well unfortunately, but this seems like something that should be possible.